DOCS-#3782: Add `using_pandas_on_python.rst` doc (#3809)
modin-project/modin: docs/developer/using_pandas_on_python.rst

pandas on Python
================

This section describes usage related documents for the pandas on Python component of Modin.

> Modin uses pandas as a primary memory format of the underlying partitions and optimizes queries

Reviewer: ```suggestion
Modin uses pandas as the primary memory format of the underlying partitions and optimizes queries
```

> ingested from the API layer in a specific way to this format. Thus, there is no need to care of choosing it

Reviewer: ```suggestion
from the API layer in a specific way to this format. Since it is a default, you do not need to specify
the pandas memory format, but we show how to explicitly set it below.
```

> but you can explicitly specify it anyway as shown below.

Reviewer: ```suggestion
```

> One of the execution engines that Modin uses is Python. This engine is sequential and it's used for the debug purposes.

Reviewer: ```suggestion
One of the execution engines that Modin uses is Python. This engine is sequential and used for the debugging.
```

> One of the execution engines that Modin uses is Python. This engine is sequential and used for the debugging.

Reviewer: ```suggestion
One of the execution engines that Modin uses is Python. This engine is sequential and used for debugging.
```
